admin/orders.php

```diff
@@ -1326,7 +1339,19 @@ function couponpopupWindow(url) {
                     $contents[] = array('align' => 'text-center', 'text' => '<a href="' . zen_href_link(FILENAME_ORDERS_INVOICE, 'oID=' . $oInfo->orders_id) . '" target="_blank" class="btn btn-info" role="button">' . IMAGE_ORDERS_INVOICE . '</a> <a href="' . zen_href_link(FILENAME_ORDERS_PACKINGSLIP, 'oID=' . $oInfo->orders_id) . '" target="_blank" class="btn btn-info" role="button">' . IMAGE_ORDERS_PACKINGSLIP . '</a>');
                     $zco_notifier->notify('NOTIFY_ADMIN_ORDERS_MENU_BUTTONS', $oInfo, $contents);
 
+                    // each contents array is drawn in a div, so this form block must be a single array element.
+                    $contents[] = ['text' =>
+                        zen_draw_form('statusUpdate', FILENAME_ORDERS, zen_get_all_get_params(['action','language']) . 'action=update_order' . (!isset($_GET['oID']) ? '&oID=' . $oInfo->orders_id : '') . '&language=' . $oInfo->language_code, 'post', '', true) . // form action uses the order language to change the session language on the update. On initial page load (from another page), $_GET['oID'] is not set, hence clause in form action
+                        zen_draw_hidden_field('listing', '1') . // identify that this form was submitted from infoBox/listing page and not the order details page (for the redirect to this same page)
+                        ($oInfo->language_code !== $_SESSION['languages_code'] ? zen_draw_hidden_field('admin_language', $_SESSION['languages_code']) : '') . // if the order language is different to the current admin language, record the admin language, to restore it in the redirect after the status update email has been sent
+                        '<label class="form-group" style="margin:8px 0">' . BOX_TOOLS_MAIL . zen_draw_checkbox_field('notify', '1', $notify_email, '', 'style="vertical-align: middle;"') . '</label>' . "<br>\n" .
+                        '<label class="form-group" for="status">' . HEADING_TITLE_STATUS . '</label>' . zen_draw_order_status_dropdown('status', $oInfo->orders_status, '', 'onChange="this.form.submit();" id="status"') . "\n" .
+                        '</form>' . "\n"];
+
                     $contents[] = array('text' => '<br>' . TEXT_DATE_ORDER_CREATED . ' ' . zen_date_short($oInfo->date_purchased));
+                    if ($_SESSION['languages_code'] !== $oInfo->language_code) {
+                        $contents[] = array('text' => zen_get_language_icon($oInfo->language_code, true) . ' ' . sprintf(TEXT_EMAIL_LANGUAGE, zen_get_language_name($oInfo->language_code, true)));
+}
```

Comment:
I wonder if it makes sense to skip the `if` part here, and just always display the language?
Or, maybe change the `if` to "if the store supports more than one language, then display it"?

Thoughts?

Reply:
As they are, in a single language system, it is always true: no icon, no complaining about the flags.
For multilanguage,
in the infoBox, in truth it is unnecessary info as nothing can be changed here. Back in time, before I figured out how to do this, this was a reminder to change the admin language before sending the email!
in details, it's useful for admin user, to add a comment in the correct language:
we are three, and all of us are fluent english/spanish and work in either admin language wihout noticing, so this is relevant.

Reply:
>  in truth it is unnecessary info as nothing can be changed here

Okay, then let's not add it if it's unnecessary.

Reply:
removed